platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

```diff
@@ -125,6 +125,62 @@ Refer to Gradle’s link:compatibility.html#kotlin[compatibility matrix] for det
+
+===== Non-abstract managed properties may no longer be attached
+**TODO, waiting on https://github.com/gradle/gradle/issues/33215 resolution**
+
+===== `DELEGATE_FIRST` closures may now prefer the delegate in some cases
+Groovy 4.0.26 has changed the behavior of closures using the `DELEGATE_FIRST` strategy.
+Dynamic lookups for properties and methods will now prefer the delegate over the owner.
+This can result in different behavior when using closures in Gradle scripts, such as certain methods not being found
+or (e.g. with `with(...) { }`) some dynamic properties taking precedence over outer properties or methods.
+
+Workarounds include using `@CompileStatic` to avoid dynamic lookups, or explicitly qualifying calls with `owner.`, `this.`, or `super.` as needed.
+
+For full clarity, in Groovy 3, the lookup order was:
+
+1. Delegate's `invokeMethod`, which chooses known methods and does no dynamic lookup.
+2. Owner's `invokeMethod`, which chooses known methods and does no dynamic lookup.
+3. Delegate's `invokeMissingMethod`, which does dynamic lookup including via properties.
+4. Owner's `invokeMissingMethod`, which does dynamic lookup including via properties.
+
+In Groovy 4, the lookup order is:
+
+1. Delegate's `invokeMethod`, which chooses known methods and does no dynamic lookup.
+2. Delegate's `invokeMissingMethod`, which does dynamic lookup including via properties.
+3. Owner's `invokeMethod`, which chooses known methods and does no dynamic lookup.
+4. Owner's `invokeMissingMethod`, which does dynamic lookup including via properties.
+
+===== Private properties and methods may be inaccessible in closures
+Closures defined in a parent class that reference its private properties or methods may no longer have access to them in subclasses.
+This is due to a https://issues.apache.org/jira/browse/GROOVY-11568[Groovy bug] that will not be resolved until Groovy 5.
+As a workaround, apply `@CompileStatic` to the class to remove the dynamic lookup.
+
+This is tracked by https://github.com/gradle/gradle/issues/32476.
+
+===== Super methods may be inaccessible from Groovy 3.x code
+Groovy 4 changed how `super` method calls are resolved at runtime.
+As a result, code compiled with Groovy 3.x may be unable to access `super` methods,
+as the code does not contain the appropriate runtime code to locate them.
+
+==== Plugins written with the Groovy DSL require Gradle >= 7.0
+
+Plugins built and published using the Groovy DSL on Gradle 9.x require Gradle 7.0 or newer to run.
+This is because Gradle 7.0 is the first release that embeds Groovy 3.0 or higher, which is required to run Groovy 4.0 compiled code.
+However, due to the major version change, some code may still behave unexpectedly.
+For full compatibility, it's recommended to run these plugins on Gradle 9.0 or later.
+
+NOTE: Plugins written using the Groovy DSL and published with Gradle 7.x or 8.x remain compatible with Gradle 5.0 and newer.
```

Comment:
```suggestion
Plugins built and published using the Groovy DSL on Gradle 9.x are only supported on Gradle 7.0 or above. Gradle 7.0 is the first release that embeds Groovy 3.0 and can run Groovy 4.0 code. 

Since Gradle 9.0 uses Groovy 4.0, some code may behave unexpectedly. For full compatibility, it's recommended to run on Gradle 9.0 or later.

NOTE: Plugins written using the Groovy DSL and published with Gradle 7.x or 8.x remain compatible with Gradle 5.0 and newer.
```